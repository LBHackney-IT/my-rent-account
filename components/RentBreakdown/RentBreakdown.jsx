import React from "react";
import PropTypes from "prop-types";

const RentBreakdown = ({ rent, benefits, toPay }) => {
  const breakdown = [
    {
      title: "Your Rent:",
      value: rent,
    },
    {
      title: "Housing Benefit Payment:",
      value: benefits,
    },
    {
      title: "What you Pay:",
      value: toPay,
    },
  ];
  return (
    <>
      <h2>Rent Breakdown</h2>
      <table className="govuk-table">
        <tbody className="govuk-table__body">
          {breakdown.map(({ title, value }) => (
            <tr key={title} className="govuk-table__row">
              <th scope="row" className="govuk-table__header">
                {title}
              </th>
              <td className="govuk-table__cell govuk-table__cell--numeric">
                Â£{value}
              </td>
            </tr>
          ))}
        </tbody>
      </table>
      <a href="/account/rent-breakdown" className="govuk-link">
        <strong>See how your rent is calculated</strong>
      </a>
    </>
  );
};

RentBreakdown.propTypes = {
  rent: PropTypes.number.isRequired,
  benefits: PropTypes.string.isRequired,
  toPay: PropTypes.string.isRequired,
};

export default RentBreakdown;
