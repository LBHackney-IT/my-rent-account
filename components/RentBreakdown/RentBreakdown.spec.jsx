import React from "react";
import { render } from "@testing-library/react";

import RentBreakdown from "./RentBreakdown";

describe("RentBreakdown component", () => {
  it("should render properly", () => {
    const { asFragment } = render(
      <RentBreakdown rent={123} toPay={321} benefits={111} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
