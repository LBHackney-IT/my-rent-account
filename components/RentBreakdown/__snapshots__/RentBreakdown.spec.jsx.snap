// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RentBreakdown component should render properly 1`] = `
<DocumentFragment>
  <h2>
    Rent Breakdown
  </h2>
  <table
    class="govuk-table"
  >
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header"
          scope="row"
        >
          Your Rent:
        </th>
        <td
          class="govuk-table__cell govuk-table__cell--numeric"
        >
          £123
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header"
          scope="row"
        >
          Housing Benefit Payment:
        </th>
        <td
          class="govuk-table__cell govuk-table__cell--numeric"
        >
          £111
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header"
          scope="row"
        >
          What you Pay:
        </th>
        <td
          class="govuk-table__cell govuk-table__cell--numeric"
        >
          £321
        </td>
      </tr>
    </tbody>
  </table>
  <a
    class="govuk-link"
    href="/account/rent-breakdown"
  >
    <strong>
      See how your rent is calculated
    </strong>
  </a>
</DocumentFragment>
`;
